File: MS.NET/Fundamentals/Platform/Persistence/textiotest.cs
using System;
using System.IO;

static class Program
{
	public static void Main(string[] args)
	{
		if(args.Length > 2)	
		{
			string name = args[0].ToUpper();
			float price = float.Parse(args[1]);	
			short stock = short.Parse(args[2]);

			TextWriter writer = new StreamWriter(
				new FileStream("product.txt", FileMode.Create));
			writer.WriteLine(name);
			writer.WriteLine(price);
			writer.WriteLine(stock);
			writer.Close();
		}
		else
		{
			TextReader reader = new StreamReader(
				new FileStream("product.txt", FileMode.Open));	
			string name = reader.ReadLine();
			float price = float.Parse(reader.ReadLine());
			short stock = short.Parse(reader.ReadLine());
			reader.Close();

			Console.WriteLine("{0} {1} {2}", name, price, stock);
		}
	}
}

File: Java/Fundamentals/Platform/Persistence/TextIOTest.java
import java.io.*;

class TextIOTest{

	public static void main(String[] args) throws Exception{
		if(args.length > 2){
			String item = args[0].toUpperCase();
			float price = Float.parseFloat(args[1]);
			short stock = Short.parseShort(args[2]);
			PrintWriter writer = new PrintWriter(
				new OutputStreamWriter(
				new FileOutputStream("product.txt")));
			writer.println(stock);
			writer.println(price);
			writer.println(item);
			writer.close();
		}else{
			BufferedReader reader = new BufferedReader(
				new InputStreamReader(
				new FileInputStream("product.txt")));
			short stock = Short.parseShort(reader.readLine());
			float price = Float.parseFloat(reader.readLine());
			String item = reader.readLine();
			reader.close();
			System.out.printf("%s %s %s%n", item, price, stock);
		}
	}
}

